plugin/src/main/java/org/opensearch/ml/action/models/UpdateModelTransportAction.java

```diff
@@ -0,0 +1,326 @@
+                .validateConnectorAccess(client, relinkConnectorId, ActionListener.wrap(hasRelinkConnectorPermission -> {
+                    if (Boolean.TRUE.equals(hasRelinkConnectorPermission)) {
+                        updateModelWithOrWithoutRelinkModelGroup(modelId, relinkModelGroupId, user, updateRequest, actionListener, context);
+                    } else {
+                        actionListener
+                            .onFailure(
+                                new MLValidationException(
+                                    "You don't have permission to update the connector, connector id: " + relinkConnectorId
+                                )
+                            );
+                    }
+                }, exception -> {
+                    log.error("Permission denied: Unable to update the connector with ID {}. Details: {}", relinkConnectorId, exception);
+                    actionListener.onFailure(exception);
+                }));
+        } else {
+            actionListener
+                .onFailure(
+                    new IllegalArgumentException("This remote does not have a connector_id field, maybe it uses an internal connector.")
+                );
+        }
+    }
+
+    private void updateRemoteOrTextEmbeddingModel(
+        String modelId,
+        MLUpdateModelInput updateModelInput,
+        MLModel mlModel,
+        User user,
+        UpdateRequest updateRequest,
+        ActionListener<UpdateResponse> actionListener,
+        ThreadContext.StoredContext context
+    ) {
+        String relinkModelGroupId = Strings.hasLength(updateModelInput.getModelGroupId()) ? updateModelInput.getModelGroupId() : null;
+        String relinkConnectorId = Strings.hasLength(updateModelInput.getConnectorId()) ? updateModelInput.getConnectorId() : null;
+
+        if (mlModel.getAlgorithm() == TEXT_EMBEDDING) {
+            if (relinkConnectorId == null) {
+                updateModelWithOrWithoutRelinkModelGroup(modelId, relinkModelGroupId, user, updateRequest, actionListener, context);
+            } else {
+                actionListener
+                    .onFailure(new IllegalArgumentException("Trying to update the connector or connector_id field on a local model"));
+            }
+        } else {
+            // mlModel.getAlgorithm() == REMOTE
+            if (relinkConnectorId == null) {
+                updateModelWithOrWithoutRelinkModelGroup(modelId, relinkModelGroupId, user, updateRequest, actionListener, context);
+            } else {
+                updateModelWithRelinkStandAloneConnector(
+                    modelId,
+                    relinkModelGroupId,
+                    relinkConnectorId,
+                    mlModel,
+                    user,
+                    updateRequest,
+                    actionListener,
+                    context
+                );
+            }
+        }
+    }
+
+    private void updateModelWithOrWithoutRelinkModelGroup(
```

Comment:
What does this function do? `relinkModelGroupId` means the new model group id? 
Suggest to avoid using new word to describe the same thing to avoid confusion. "Relink" may confuse as we don't have concept to "link" model to model group. We just add/register a model to a model group.

Reply:
What about `registerNewModelGroupId`?